src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -559,26 +589,39 @@ public void Write(ArraySegment<byte> data)
             {
                 if (_autoChunk)
                 {
-                    if (data.Count == 0)
-                    {
-                        if (firstWrite)
-                        {
-                            Flush();
-                        }
-                        return;
-                    }
-                    WriteChunked(data);
+                    WriteChunked();
                 }
                 else
                 {
-                    CheckLastWrite();
-                    Output.Write(data);
+                    Write();
                 }
             }
             else
             {
                 HandleNonBodyResponseWrite();
+                FlushIfNeeded();
+            }
+
+            void Write()
```

Comment:
Don't know yet; though it would need to take `data` as a param.

Also means all the `Write(Async)XXX` functions disappear from the function list as they are all internal to their respective functions.